File: finalThroughPJT_server/src/main/java/com/ssafy/cafe/model/service/UserCustomService.java
package com.ssafy.cafe.model.service;

import java.util.List;

import com.ssafy.cafe.model.dto.UserCustom;

public interface UserCustomService {
	
	/**
     * 사용자 custom + 즐겨찾기 메뉴를 DB에 저장한다.
     * 
     * @param UserCustom
     */
	public void addUserCustomProd(UserCustom userCustom);
	
	/**
	 * id에 해당하는 사용자 custom 상품 정보를 삭제한다.
	 * @param customId
	 */
	public void removeUserCustomProd(Integer customId);
	
	/**
	 * id에 해당하는 사용자 custom 상품 정보를 반환한다.
	 * @param customId
	 * @return
	 */
	UserCustom selectUserCustom(Integer customId);
	
	/**
	 * 모든 사용자 custom 상품을 반환한다.
	 * @return
	 */
	List<UserCustom> selectUserCustomList();
	
	/**
	 * 사용자 id에 해당하는 사용자 custom 상품 정보를 반환한다.
	 * @param userId
	 * @return
	 */
	List<UserCustom> selectUserCustomWithUserId(String userId);
	
}

File: finalCoffeeProject/app/src/main/java/com/ssafy/cafe/src/main/network/service/UserCustomService.kt
package com.ssafy.cafe.src.main.network.service

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.ssafy.cafe.src.main.dto.UserCustom
import com.ssafy.cafe.util.RetrofitCallback
import com.ssafy.cafe.util.RetrofitUtil
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class UserCustomService {
    private val TAG = "UserCustomService_싸피"
    fun insertCustomMenu(userCustom: UserCustom, callback: RetrofitCallback<Boolean>){
        RetrofitUtil.customService.insert(userCustom).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200){
                    if(res == true){
                        callback.onSuccess(response.code(), res)
                    }else{
                        callback.onFailure(response.code())
                    }
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    fun getCustomWithUserId(userId:String) : LiveData<List<UserCustom>> {
        val responseLiveData : MutableLiveData<List<UserCustom>> = MutableLiveData()
        val getUserCustomRequest: Call<List<UserCustom>> = RetrofitUtil.customService.getCustomWithUserId(userId)

            getUserCustomRequest.enqueue(object : Callback<List<UserCustom>>{
                override fun onResponse(
                    call: Call<List<UserCustom>>,
                    response: Response<List<UserCustom>>
                ) {
                    val res = response.body()
                    if(response.code() == 200){
                        if(res!=null){
                            responseLiveData.value = res
                        }
                    }else{
                        Log.d(TAG, "onResponse: ")
                    }
                }

                override fun onFailure(call: Call<List<UserCustom>>, t: Throwable) {
                    Log.d(TAG, "onFailure: ")
                }
            })

        return responseLiveData
    }

    fun deleteCustomMenu(customId:Int, callback:RetrofitCallback<Boolean>) {
        RetrofitUtil.customService.delete(customId).enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200){
                    if(res != null){
                        callback.onSuccess(response.code(), res)
                    }
                }else{
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

}